syntax = "proto3";


package noteworthy_hub;


message ReservationRequest {
    // mydomain.noteworthy.im
    string domain = 1;
    // wireguard pub key
    string pub_key = 2;
    // auth_key / invitation code
    string auth_code = 3;
}

message ReservationResponse {
    // wireguard endpoint ip
    string link_endpoint = 1;
}

message PeeringRequest {
    string wg_pubkey = 1;
    string auth_token = 2;
}

message PeeringResponse {
    string hub_wg_pubkey = 1;
    string hub_wg_endpoint = 2;
}